sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -705,6 +705,7 @@ class Dataset[T] private[sql](
         LogicalRDD(
           logicalPlan.output,
           internalRdd,
+          None,
```

Comment:
why not set the logical plan here?

Reply:
That is a silly missing point. Thanks for finding this out!

Reply:
#37167